components/board/customlinux/board.go

```diff
@@ -0,0 +1,146 @@
+//go:build linux
+
+// Package customlinux implements a board running linux
+// This is an Experimental package
+package customlinux
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"periph.io/x/host/v3"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/board/genericlinux"
+	"go.viam.com/rdk/resource"
+)
+
+func init() {
+	if _, err := host.Init(); err != nil {
+		golog.Global().Debugw("error initializing host", "error", err)
+	}
+
+	RegisterCustomBoard("customlinux")
+}
+
+// customLinuxBoard wraps the genericlinux board type so that both can implement their own Reconfigure function.
+type customLinuxBoard struct {
+	*genericlinux.SysfsBoard
+}
+
+// RegisterCustomBoard registers a sysfs based board using the pin mappings.
+func RegisterCustomBoard(modelName string) {
+	resource.RegisterComponent(
+		board.API,
+		resource.DefaultModelFamily.WithModel(modelName),
+		resource.Registration[board.Board, *Config]{
+			Constructor: createNewBoard,
+		})
+}
+
+func createNewBoard(
+	ctx context.Context,
+	_ resource.Dependencies,
+	conf resource.Config,
+	logger golog.Logger,
+) (board.Board, error) {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return nil, err
+	}
+
+	pinDefs, err := parseBoardConfig(newConf.PinConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	gpioMappings, err := genericlinux.GetGPIOBoardMappingFromPinDefs(pinDefs)
+	if err != nil {
+		return nil, err
+	}
+
+	boardConfig := genericlinux.Config{
+		I2Cs:              newConf.I2Cs,
+		SPIs:              newConf.SPIs,
+		Analogs:           newConf.Analogs,
+		DigitalInterrupts: newConf.DigitalInterrupts,
+	}
```

Comment:
This needs to be passed its name and a logger, or else it will have nil pointer dereferences once something tries to log or call its name.

Reply:
Oooo, you can initialise it with the looger and name as a struct, and then you have something to tidy up with your reconfigure function.

Reply:
the underlying genericlinux board gb should be initialized with the name and logger right?

Reply:
yep, you're passing those through conf, but this wrapper sturct also needs to know those two pieces of information, otherwise calling resource.Name on the wrapping resource causes a nil pointer dereference.